policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/stream/config/AkkaRunnableGraph.java

```diff
@@ -70,68 +72,64 @@ KafkaProducerService kafkaProducerService(
+                                //check to see if the first service request threw an exception and if so deal with it
+                                .thenApply(modifiedResource -> modifiedResource.chain(messageAndModifiedRequest.second().getAuditErrorMessage()))
+                                .thenApply(response -> Pair.create(messageAndModifiedRequest.first(), response))
+                )
 
-                // Now we build the producer message for kafka, adding the original committable
-                // Filtered out resources use a pass through message, they aren't sent downstream but are committed upstream
-                .map((Pair<CommittableMessage<String, PolicyRequest>, Optional<Optional<PolicyResponse>>> messageAndResponse) -> {
-                    Committable committable = messageAndResponse.first().committableOffset();
-                    ConsumerRecord<String, PolicyRequest> record = messageAndResponse.first().record();
-                    return messageAndResponse.second()
-                            // If this is a proper request
-                            .map(maybeResponse -> maybeResponse
-                                    // If the resource was not redacted
-                                    .map(response -> ProducerMessage.single(new ProducerRecord<>(outputTopic.getName(), record.partition(), record.key(), response, record.headers()), committable))
-                                    // If the resource was redacted
-                                    .orElse(ProducerMessage.passThrough(committable)))
-                            // If this is a start or end of stream
-                            .orElse(ProducerMessage.single(new ProducerRecord<>(outputTopic.getName(), record.partition(), record.key(), null, record.headers()), committable));
-                })
+                // Build producer message, copying the partition, keeping track of original message
+                .map((Pair<CommittableMessage<String, PolicyRequest>, AuditablePolicyRecordResponse> messageAndResponse) -> {
+                    ConsumerRecord<String, PolicyRequest> requestRecord = messageAndResponse.first().record();
+                    Committable commitable = (Committable) messageAndResponse.first().committableOffset();
+                    return Optional.ofNullable(messageAndResponse.second().getAuditErrorMessage())
+                            // Found an application error, produce an error message to be sent to the Audit service
+                            .map(audit -> ProducerMessage.single(
+                                    new ProducerRecord<>(errorTopic.getName(), requestRecord.partition(), requestRecord.key(),
+                                            SerDesConfig.errorValueSerializer().serialize(null, audit), requestRecord.headers()),
+                                    (Committable) messageAndResponse.first().committableOffset()))
```

Comment:
Can we not use the committable variable here?
```suggestion
                                    committable))
```

Reply:
Did the change and corrected the spelling